From 7e41694a353757def3ae28e3cc7e86b83f22dc7e Mon Sep 17 00:00:00 2001
From: Jason Ekstrand <jason.ekstrand@intel.com>
Date: Mon, 8 Jun 2015 14:20:42 -0700
Subject: [PATCH] Fix PNG

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fdd9d17..de003cc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -66,8 +66,8 @@ endif ()
 include_directories(${ZLIB_INCLUDE_PATH})
 
 # libpng
-find_path(PNG_INCLUDE_PATH	libpng.h)
-find_library(PNG_LIBRARY	png)
+find_path(PNG_INCLUDE_PATH	png.h)
+find_package(PNG)
 
 if (NOT PNG_INCLUDE_PATH OR NOT PNG_LIBRARY)
 	message(STATUS "System version of libpng not found, using external/libpng")
-- 
2.6.1

